src/coreclr/gc/gc.cpp

```diff
@@ -23963,11 +23989,16 @@ void gc_heap::background_process_mark_overflow_internal (uint8_t* min_add, uint8
             while (seg)
             {
 #ifdef USE_REGIONS
-                if (heap_segment_overflow_p (seg))
+                bool ephemeral = (heap_segment_gen_num (seg) < max_generation);
+                if ((heap_segment_overflow_p (seg) && !ephemeral) || (!concurrent_p && ephemeral))
```

Comment:
as mentioned you don't need to check for ephemeral here - we are already only going through eph when it's concurrent